### PR TITLE
Replace get_bus with the class DBus 

### PR DESCRIPTION
**Pull request by poncovka**
We should call static methods of DBus class instead of get_bus. The
class DBus keeps one instance of the DBus connection that is created
on demand. It also provides methods for easy publishing and creating
proxies.

We should never keep our own instances of the DBus connection unless
we have a very good reason for it.

The module pyanaconda.dbus.dbus_constants was renamed
to pyanaconda.dbus.constants.

The PYTHONPATH should be only modified in run_boss_locally, not
replaced, so we can change it and add our local repositories.

### CODE REVIEW

File: pyanaconda/dbus/__init__.py [hunk 1]
```diff
@@ -1,21 +1,86 @@
...
-    if bus_address:
-        return pydbus.connect(os.environ["DBUS_STARTER_ADDRESS"])
-    else:
-        log.info("DBUS_STARTER_ADDRESS not defined, using system bus")
+    _connection = None
+
+    @staticmethod
+    def get_connection():
+        """Returns a DBus connection."""
+        if not DBus._connection:
```
Comment (M4rtinK) [inline pyanaconda/dbus/__init__.py]: I wonder if there is a chance for a race condition ? Eq. two requests getting two different connections, with the last one being cached.

File: pyanaconda/dbus/__init__.py [hunk 2]
```diff
@@ -1,21 +1,86 @@
...
+
+        if bus_address:
+            log.info("Connecting to dbus at %s.", bus_address)
+            return pydbus.connect(bus_address)
+
+        log.info("Connecting to system dbus.")
         return pydbus.SystemBus()
 
+    @staticmethod
+    def publish(bus_name, *objects):
```
Comment (M4rtinK) [inline pyanaconda/dbus/__init__.py]: So I guess this makes it possible to publish multiple class instances under the same dbus_name ? I wonder what's the use-case for that ?

File: pyanaconda/dbus/__init__.py (hunk 1 above)
Comment (poncovka) [reply pyanaconda/dbus/__init__.py]: I don't think that could be a problem. The cached connection is just recommended to use, but it is not required.

**Comment by poncovka** [comment]
Reworking to add support for object paths.

**Comment by poncovka** [comment]
Updated.

**Comment by poncovka** [comment]
Updated.

File: pyanaconda/dbus/constants.py [hunk 3]
```diff
@@ -17,6 +17,8 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 
+from pydbus.auto_names import auto_object_path
```
Comment (jkonecny12) [inline pyanaconda/dbus/constants.py]: It would be nice to have comment describing why we need it here. What's the reason.

**Pull request merged**